practices/javascript-styling.md

```diff
@@ -0,0 +1,35 @@
+# Javascript dependent styling
+
+A class name of `.js` on the document root element (normally `html`) will be used to bootstrap JS-only visual styles. We do this to prevent the visual flicker and browser reflow that may occur when JavaScript applies CSS styles to a component after the page has loaded.
+
+## Scripts
+
+The `.js` class must be added via a micro (but blocking) script, placed as far up in the `<head>` of the document as possible.
```

Comment:
```suggestion
The `.js` class must be added via a micro (but blocking) script, placed as far up in the `<head>` of the document as possible.

The blocking nature of the script prevents any visual flickering or browser reflows, unlike other techniques where JavaScript may cause CSS styles to be applied after the page has rendered.
```

Is this what we are driving at?

Reply:
Yep, exactly this (at least this is also my assumption, @sonniesedge wrote it!)

Reply:
Yup, that's exactly it. Although I can't remember writing this. 

If it's a success I shall claim credit. If not, Alex wrote it and blamed me.